cmake/ctest_script.cmake

```diff
@@ -482,7 +483,7 @@ function(update_gismo ug_ucount)
 	WORKING_DIRECTORY ${CTEST_SOURCE_DIRECTORY}/extensions/${submodule})
       ctest_update(SOURCE ${CTEST_SOURCE_DIRECTORY}/extensions/${submodule} RETURN_VALUE ug_upd_sm)
       if (${ug_upd_sm} GREATER 0)
-	math(EXPR updcount "${updcount} + ${ug_upd_sm}")
+	math(EXPR ${ug_ucount} "${${ug_ucount}} + ${ug_upd_sm}")
```

Comment:
You have no write access to the variable/parameter at **PARENT_SCOPE** via `math(EXPR ${ug_ucount} "...")`.
Only `set(${ug_ucount} ... PARENT_SCOPE)` has write access to the variable/parameter at parent scope.
(And only if it is a parameter (a new variable) at parent scope, otherwise it seams we have no chance to write to it, but well, we do not want to override variables of the PARENT_SCOPE by accident, like using the same name twice).

Think of it like 2 separate variables, one is a copy you can access with `${var}`, the other is a reference you can't access normally, but the `set(${var} ... PARENT_SCOPE)` can - the `set(var ...)` will break it up. So every write to `${var}` modifies only the copy. And nearly every read of `${var}` is only one of the copy. Nearly, because a `set(${var} ${var} PARENT_SCOPE)` has the reference for all parameters that are not defined local (with `set(var ...)`) on both sides and doesn't do anything in our case. - Well, because our ${var} gets to be a variable with the name par (if function f was called as f(par)) it's getting a bit complicate and we should look not to use names twice, write isn't the problem there, it's the read, that should be the parent before a write and the local after a write and is a bit brain dizzying. 

Therefore on the local scope it seams to work - I think you tested it with the out-commented `#if (${ug_ucount} GREATER 0) endif()`, but on the parent scope, it's not known that something changed - or that the variable `updcount` has a value. So we result in an error saying **`if( GREATER 0)` doesn't work**.

By the way, you do not include the result value of `ctest_update(SOURCE ${CTEST_SOURCE_DIRECTORY} RETURN_VALUE updcount)`. OK, seams you thought that it is the same parameter(name), but not the same scope.

Reply:
Probably macros work more the way we want, so I changed to a macro now!

Reply:
Macros do also need the PARENT_SCOPE if I remember correct to get out. All variables are just copies. My girlfriend say's I should stop now - I'm still at home and **this** keeps me busy.